presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -97,6 +97,18 @@ public void testCreateFunctionInvalidSemantics()
                 ".*CREATE FUNCTION body cannot contain aggregations, window functions or grouping operations:.*");
     }
 
+    @Test
+    public void testCreateFunctionExternal()
+    {
+        assertQuerySucceeds("CREATE FUNCTION testing.test.foo(x varchar) RETURNS varchar LANGUAGE JAVA EXTERNAL");
+        assertQuerySucceeds("CREATE FUNCTION testing.test.foo(x int) RETURNS bigint LANGUAGE JAVA EXTERNAL NAME foo_from_another_library");
+        assertQuerySucceeds("CREATE FUNCTION testing.test.foo(x bigint) RETURNS bigint LANGUAGE JAVA EXTERNAL NAME \"foo.from.another.library\"");
+        assertQuerySucceeds("CREATE FUNCTION testing.test.foo(x smallint) RETURNS bigint LANGUAGE JAVA EXTERNAL NAME 'foo.from.another.library'");
+        assertQuerySucceeds("CREATE FUNCTION testing.test.foo(x double) RETURNS double LANGUAGE \"JAVA\" EXTERNAL");
+
+        assertQueryFails("CREATE FUNCTION testing.test.foo(x varchar) RETURNS varchar LANGUAGE UNSUPPORTED EXTERNAL", "Catalog testing does not support functions implemented in language UNSUPPORTED");
```

Comment:
I think the first 2 are valid syntax.

Reply:
Sure, let's add the test cases then.

Reply:
I think the third one is also valid syntax though it might not be valid semantically. The order of validation should be:
1) whether the function namespace allows create
2) whether the function namespace supports the provided language
3) whether the provided language in specified function namespace support the given function body

Without checking the capability of the specific function namespace, all above mentioned are valid syntax.

Reply:
Sure, that all make sense to me. My ask is to add those test cases here so that we can see the query are passing as expected. If there are any behavior changes in the future, the test will catch that.